File: backend/app/schemas/opportunity.py
"""商机相关 Schema。"""
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict


class OpportunityCreate(BaseModel):
    customer_id: int
    name: str
    product: str | None = None
    budget: float | None = None
    stage: str = "contact"
    expected_close_date: date | None = None
    owner_id: int | None = None
    remark: str | None = None


class OpportunityOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    customer_id: int
    name: str
    product: str | None = None
    budget: float | None = None
    stage: str
    expected_close_date: date | None = None
    owner_id: int | None = None
    remark: str | None = None
    created_at: datetime
    updated_at: datetime

File: backend/src/main/java/com/crm/entity/Opportunity.java
package com.crm.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 商机。
 */
@Data
@TableName("opportunities")
public class Opportunity {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long customerId;
    private String name;
    private String inquiry;
    private String product;
    private BigDecimal budget;
    private String stage;
    private LocalDate expectedCloseDate;
    private Long ownerId;
    private String remark;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
